### DISCUSSION TITLE
Adding general C++20 support

### DISCUSSION THREAD

**Discussion by Neuromancer2701** [lefticus/cpp_starter_project#177] [Q&A] [answered]
I am trying to do a C++20 Modules test.

I have figured out a way in CMAKE to add the module flag.
But I can't seem to get the project to switch over to C++20 to recognize it.

What would be the easier and/or the best way to switch to C++20?

See this fork for some of the stuff I have tried:
https://github.com/Neuromancer2701/test_conan_cpp_project

**Comment by ddalcino** [comment, marked as answer]
From your own root level CMakeLists.txt:

```cmake
target_compile_features(project_options INTERFACE cxx_std_20)
```

This is the correct way to set the `-std=c++20` flag in a CMake project (see [CMake docs](https://cmake.org/cmake/help/latest/manual/cmake-compile-features.7.html#requiring-language-standards)). There is a CMAKE_CXX_FLAGS variable that you can play with, but it's not the accepted "best practice" way to do things.

If you want to confirm that the flag is set properly, you can use `-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON` as an argument to `cmake` during the configure step. This will turn on verbose mode, which will print every compile command so that you can see what flags are being passed to the compiler:
```bash
cmake -S . -B ./build -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON
```

I don't know anything about adding C++20 module support; last time I checked, there weren't any compilers that had fully implemented the feature yet. You may need to check that your compiler supports this feature.
